### PR TITLE
Allows for serializing of public/private keys to raw bytes

### PR DESCRIPTION
**Pull request by thanethomson**
As per #25, this code adds the ability to get the underlying raw bytes comprising an `EphemeralPublic` instance.

Let me know if the IDE auto-formatting should be removed and I'll revert to the original formatting of the code.

### CODE REVIEW

File: src/x25519.rs [hunk 1]
```diff
@@ -20,22 +20,30 @@ use curve25519_dalek::constants::ED25519_BASEPOINT_TABLE;
...
 impl From<[u8; 32]> for EphemeralPublic {
     /// Given a byte array, construct an x25519 `EphemeralPublic` key
     fn from(bytes: [u8; 32]) -> EphemeralPublic {
         EphemeralPublic(MontgomeryPoint(bytes))
     }
+}
 
+impl EphemeralPublic {
+    /// Utility function to convert this public key to its raw bytes.
+    pub fn to_bytes(&self) -> [u8; 32] {
```
Comment (Liamsi) [inline src/x25519.rs]: Consider Tony's comment here: https://github.com/dalek-cryptography/x25519-dalek/issues/25#issuecomment-456439837

Comment (hdevalence) [reply src/x25519.rs]: I'm not sure if there's a good reason to have both `as_bytes` and `to_bytes`, maybe we could just have `as_bytes`?

File: src/x25519.rs [hunk 2]
```diff
@@ -63,6 +77,22 @@ impl EphemeralSecret {
...
+        self.0.to_bytes()
+    }
+}
+
+impl From<[u8; 32]> for EphemeralSecret {
+    /// Given a byte array, construct an `EphemeralSecret`.
+    fn from(bytes: [u8; 32]) -> EphemeralSecret {
+        EphemeralSecret(clamp_scalar(bytes))
+    }
```
Comment (hdevalence) [inline src/x25519.rs]: If you can extract the secret bytes from an `EphemeralSecret`, and then feed them into a new `EphemeralSecret` object, it's not guaranteed to be ephemeral any more. So I'm not sure this is a good change.

**Comment by hdevalence** [comment]
The `EphemeralPublic::as_bytes` was supposed to be there but was missed, so I did an 0.4.4 release immediately that fixes it.

**Comment by thanethomson** [comment]
No longer necessary now with the latest release - thanks very much @hdevalence! 👍

**Pull request closed without merging**